nav2_system_tests/src/recoveries/backup/test_backup_recovery_node.cpp

```diff
@@ -72,10 +72,9 @@ TEST_P(BackupRecoveryTestFixture, testBackupRecovery)
   success = backup_recovery_tester->defaultBackupRecoveryTest(target_dist, tolerance);
 
   // if intentionally backing into an obstacle, should fail.
-  // TODO(stevemacenski): uncomment this once note below completed.
-  // if (target_dist < -1.0) {
-  //   success = !success;
-  // }
+  if (target_dist < -1.0) {
```

Comment:
Need to add a similar collision test for spin recovery as well

Reply:
Hi Sarthak, Forgive me I am not quite familiar with writing tests with Gazebo. How do you suggest I force the robot into a situation where the spin action would be aborted?

Reply:
@SteveMacenski , @naiveHobo and I spoke, and we are both looking for some input in how we create a spin collision in an automated way.